planner/core/integration_partition_test.go

```diff
@@ -1136,3 +1136,31 @@ func TestIssue27532(t *testing.T) {
 	tk.MustQuery(`select * from t2`).Sort().Check(testkit.Rows("1 1 1 1", "2 2 2 2", "3 3 3 3", "4 4 4 4"))
 	tk.MustExec(`drop table t2`)
 }
+
+func TestIssue32516(t *testing.T) {
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("create database issue_32516")
+	defer tk.MustExec(`drop database issue_32516`)
+	tk.MustExec("use issue_32516")
+	tk.MustExec("set @@tidb_partition_prune_mode=dynamic;")
+	tk.MustExec("CREATE TABLE pt (" +
+		"  `id` bigint(20) DEFAULT NULL," +
+		"  a int," +
+		"  KEY (a))" +
+		"PARTITION BY RANGE(id)" +
+		"(PARTITION p1 VALUES LESS THAN (4)," +
+		" PARTITION p2 VALUES LESS THAN (8)," +
+		" PARTITION p3 VALUES LESS THAN (12)," +
+		" PARTITION p4 VALUES LESS THAN (MAXVALUE));")
+	tk.MustExec("insert into pt values (1,1),(2,2),(3,3),(4,4),(5,5),(6,6),(7,7),(8,8),(9,9),(10,10),(11,11),(12,12),(13,13),(14,14),(15,15),(16,16);")
+	tk.MustQuery("SELECT COUNT(*) FROM (SELECT * FROM pt WHERE a BETWEEN 1 AND 15 LIMIT 2) as t;").Check(testkit.Rows("2"))
+	tk.MustQuery("EXPLAIN SELECT * FROM pt WHERE a BETWEEN 1 AND 15 LIMIT 2;").Check(testkit.Rows(
```

Comment:
Maybe we can use `explain format = 'brief' select ...` here. In this way, plan IDs don't appear(`Limit_9` -> `Limit`). Some Changes in planner may change plan IDs but don't change the plan itself. We don't need to modify explain results in the test cases if we use brief mode.